### PR TITLE
Update winit crate to 0.21.0

### PR DESCRIPTION
**Pull request by doublej472**
- [x] Tested on all platforms changed
- [x] Compilation warnings were addressed
- [x] `cargo fmt` has been run on this branch
- [x] `cargo doc` builds successfully
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality

`winit` version 0.21.0 fixes a bug in another project at https://github.com/alacritty/alacritty/issues/3191.

This builds correctly, and all checks seem to pass. [`winit`](https://github.com/rust-windowing/winit/compare/627a127f1b96608649eee5ac474bf762daffbc3a..28f0eb598d8a309f528caf540751a28ec3c68ace) should have only 2 breaking changes, and they both seem to not affect this crate.

I've done some minor testing with Alacritty using Wayland on Linux, and it seems to work fine, plus the fixed bug.

Let me know if there is anything else I need to do!

### CODE REVIEW

**Comment by doublej472** [comment]
It looks like there is a dropped target that affects CI, the `armv7-apple-ios` target. Support for it in the winit crate has been [removed here](https://github.com/rust-windowing/winit/pull/1433).

**Comment by goddessfreya** [comment]
Yeah. Can you remove the CI target from here, bump glutin's version to 0.23 and update the readme?

**Comment by doublej472** [comment]
@goddessfreya I've gone ahead and made those changes.

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Join us in any of these:
 
 ## Usage Examples
 
-Warning: these are examples for master. For the latest released version, 0.22, view [here.](https://github.com/rust-windowing/glutin/tree/447f3526dcf90a460d52afefd0b29eb2ed7f87f3)
+Warning: these are examples for master. For the latest released version, 0.23, view [here.](https://github.com/rust-windowing/glutin/tree/2bcb198b85b74fea0c864f85f3a72c3e27f07337)
```
Comment (goddessfreya) [inline README.md]: This hash will change when we squash commit. Leave this one for me to do in a separate commit.

Comment (doublej472) [reply README.md]: I was wondering about that, I've amended the commit.
